cli/cli.py

```diff
@@ -0,0 +1,206 @@
+    f'populationgenomics/analysis-runner/{BRANCH}/cli/servermap.json'
+)
+# Make this None if you want to support all organisations
+SUPPORTED_ORGANISATIONS = {'populationgenomics'}
+
+
+@click.command(
+    help='CLI for the analysis runner - a CPG service for running analysis from '
+    'some GitHub repository (at a specific commit). The parameters are used to form a '
+    'POST request, sent to a server based on the --dataset parameter you provide.'
+)
+@click.option(
+    '--dataset',
+    required=True,
+    help='The dataset name, which determines which '
+    'analysis-runner server to send the request to',
+)
+@click.option(
+    '--output-dir',
+    '-o',
+    required=True,
+    help='The output directory of the run, MUST start with gs://',
+)
+@click.option(
+    '--repository',
+    '--repo',
+    help='The URI of the repository to run, must be approved by the appropriate server.'
+    ' Default behaviour is to find the repository of the current working'
+    ' directory with `git remote get-url origin`',
+)
+@click.option(
+    '--commit',
+    '--hash',
+    help='The hash of commit to run, default behaviour is to ' 'use the current commit',
+)
+@click.option(
+    '--description',
+    required=True,
+    help='Description of job, otherwise defaults to: "$USER FROM LOCAL: $REPO@$COMMIT"',
+)
+@click.argument('script', nargs=-1)
+def main(dataset, output_dir, script, description, commit=None, repository=None):
+    """
+    Main function that drives the CLI.
+    The parameters are provided automatically by @click.
+    """
+
+    if repository is not None and commit is None:
+        raise Exception(
+            "You must supply the '--commit <SHA>' parameter "
+            "when specifying the '--repository'"
+        )
+
+    _repository = repository or _get_default_remote()
+    _commit_ref = commit or _get_commit_ref_of_current_repository(
+        used_custom_repository=repository is not None
+    )
+    _url = _get_url_from_dataset(dataset)
+    _token = _get_google_auth_token()
+    _script = script or ['main.py']
+
+    logging.info('Submitting %s@%s for dataset "%s"', _repository, _commit_ref, dataset)
+
+    response = requests.post(
+        _url,
+        json={
+            'output': output_dir,
+            'repo': _repository,
+            'commit': _commit_ref,
+            'script': _script,
+            'description': description,
+        },
+        headers={'Authorization': f'Bearer {_token}'},
+    )
+    try:
+        response.raise_for_status()
+        logging.info('Request submitted successfully: %s', response.text)
+    except requests.HTTPError as e:
+        logging.critical(
+            'Request failed with status %s: %s\nFull response: %s',
+            response.status_code,
+            str(e),
+            response.text,
+        )
+
+
+def _get_google_auth_token() -> str:
+    # https://stackoverflow.com/a/55804230
+    # command = ['gcloud', 'auth', 'print-identity-token']
+
```

Comment:
```suggestion
```

Reply:
I'd prefer to keep this here, I think provides clear context for the motivation of the function.